src/identity.js

```diff
@@ -436,93 +436,81 @@ export class Identity extends EventEmitter {
+                    throw new SDKError('HasSession failed', sessionData.error);
                 }
                 this._maybeSetVarnishCookie(sessionData);
                 this._emitSessionEvent(this._session, sessionData);
             };
-
-            if (typeof autologin !== 'boolean') {
-                const [type, value] = inspect(autologin);
-                return reject(new SDKError(`Parameter 'autologin' must be boolean, was: "${type}:${value}"`));
-            }
-
-            try {
+            const _getSession = async () => {
+                let sessionData = null;
                 if (this._enableSessionCaching) {
                     // Try to resolve from cache (it has a TTL)
-                    const cachedData = this.cache.get(HAS_SESSION_CACHE_KEY);
-                    if (cachedData) {
-                        postProcess(cachedData);
-                        return resolve(cachedData);
-                    }
+                    sessionData = this.cache.get(HAS_SESSION_CACHE_KEY);
                 }
-
-                let data = null;
-                if (this._sessionService) {
+                if (!sessionData && this._sessionService) {
                     try {
-                        data = await this._sessionService.get('/session');
-                    } catch (err) {	
+                        sessionData = await this._sessionService.get('/session');
+                    } catch (err) {
                         if (this.siteSpecificLogout) {
                             // Don't fallback to other sources for user session lookup
                             throw err;
                         }
 
-                        // The session-service returns 400 if no session-cookie is sent in the	
-                        // request. This will be the case if the user hasn't logged in since the	
-                        // site switched to using the session-service. If the request contains a	
-                        // session-cookie but no session is found (return code will be 404), then we	
-                        // *should* throw an exception and *not* fall through to spid-hassession	
-                        if (err.code !== 400) {	
-                            this.emit('error', err);	
-                            return reject(new SDKError('HasSession failed', err));	
-                        }	
-                        data = null;	
+                        // The session-service returns 400 if no session-cookie is sent in the
+                        // request. This will be the case if the user hasn't logged in since the
+                        // site switched to using the session-service. If the request contains a
+                        // session-cookie but no session is found (return code will be 404), then we
+                        // *should* throw an exception and *not* fall through to spid-hassession
+                        if (err.code !== 400) {
+                            throw err;
```

Comment:
> Also looks like tests coverage has drop a bit

Test coverage hasn't actually changed. But since I changed these lines, that were previously not covered by the tests, the "patch diff" has uncovered lines as well. I'll see if I can add some test for this case.

Reply:
I tried to add a test case, but since I'm not really familiar with the evolution in the flow of different services, I'm not sure whether it makes much sense. Can you have a look at it?

Reply:
Yes, I will 👍